c/parallel/src/jit_templates/templates/operation.h

```diff
@@ -13,20 +13,61 @@
 #ifndef _CCCL_C_PARALLEL_JIT_TEMPLATES_PREPROCESS
 #  include <cuda/std/cstddef>
 #  include <cuda/std/type_traits>
+#  include <cuda/std/utility>
 
 #  include <cccl/c/types.h>
 #endif
 
 #include "../mappings/operation.h"
 #include "../mappings/type_info.h"
 
+// Helper to define the C function pointer type: void (*)(void*..., void*)
+template <typename Seq>
+struct MakeVoidPtrFuncPtr;
+template <cuda::std::size_t... Is>
+struct MakeVoidPtrFuncPtr<cuda::std::index_sequence<Is...>>
+{
+  // Helper type that is always void*
+  template <cuda::std::size_t>
+  using void_ptr_t = void*;
+  // Define the function pointer type
+  using type = void (*)(void_ptr_t<Is>..., void*); // Last void* is for result
+};
+
+// Helper to define the C function pointer type for stateful ops: void (*)(void* state, void*..., void*)
+template <typename Seq>
+struct MakeVoidPtrStatefulFuncPtr;
+template <cuda::std::size_t... Is>
+struct MakeVoidPtrStatefulFuncPtr<cuda::std::index_sequence<Is...>>
+{
+  template <cuda::std::size_t>
+  using void_ptr_t = void*;
```

Comment:
Since `void_ptr_t` applies to input arguments, which are read-only, should be be `using void_ptr_t = const void *l`?

Reply:
And ditto for `void_ptr_t` in `MakeVoidPtrFuncPtr`